packages/dbml-core/src/parse/ANTLR/ASTGeneration/postgres/PostgresASTGen.js

```diff
@@ -804,6 +804,36 @@ export default class PostgresASTGen extends PostgreSQLParserVisitor {
             cmd.value.endpoints[0].schemaName = schemaName;
             this.data.refs.push(cmd.value);
             break;
+          case TABLE_CONSTRAINT_KIND.UNIQUE:
+          case TABLE_CONSTRAINT_KIND.INDEX:
+            kind = cmd.kind;
+            if (cmd.value.pk || cmd.value.unique) {
+              let set_prop = cmd.value.unique ? (obj) => obj["unique"] = true : (obj) => obj["pk"] = true;
+              const table = findTable(this.data.tables, schemaName, tableName);
+              if (!table){
+                break;
+              }
+              if (cmd.value.columns.length == 1) {
+                // We are in the single column case, we can update the column as well
+                let key = cmd.value.columns[0].value;
+                let field_to_update = table.fields.find(f => f.name === key);
+                // field might have been deleted
+                if (field_to_update != null) {
+                  set_prop(field_to_update);
+                }
+              }
+              let index = table.indexes.find(idx => { idx.name == cmd.value.name });
+              if (!index) {
+                index = new Index({
+                  name: cmd.value.name,
+                  columns: cmd.value.columns,
+                });
+                table.indexes.push(index);
+              }
+              set_prop(index);
+              index.columns = cmd.value.columns;
+            }
+            break;
```

Comment:
Hi I suggest to update the logic to the below, the main points are:
* Generating both fields settings and indexes for the same constraint is not recommended, because when re-exported to SQL, it'll generate both a column constraint in CREATE TABLE column definition and a CREATE INDEX statement. Therefore. we should only choose one.
* IMO, using pk/unique field settings to describe SQL constraints is the better because it more inline with concept of SQL constraints, while indexes fit more with CREATE INDEX.. statements. However, it has two downsides:
  * It can only used for single column constraint -> I agree that for multi-column constraint, using to indexes is our only choice.
  * We can't name the pk/unique constraint in field settings -> For now, I recommend that for single-column named constraint, we still import it as a field property. In the future, we intend to add the ability to name field pk/unique property through this syntax in dbml. Please let me know what you think about this:
  ```
  table a {
    id int [pk: "pk_constraint_name"]
  }
  ...
  ```
```suggestion
          case TABLE_CONSTRAINT_KIND.UNIQUE:
          case TABLE_CONSTRAINT_KIND.INDEX: {
            if (!(cmd.value.pk || cmd.value.unique)) break;

            // eslint-disable-next-line prefer-destructuring
            kind = cmd.kind;

            const table = findTable(this.data.tables, schemaName, tableName);

            if (!table) break;

            // multi column constraint -> need to create new index
            if (cmd.value.columns.lenght > 1) {
              const index = new Index({
                name: cmd.value.name,
                columns: cmd.value.columns,
              });

              index[cmd.value.unique ? 'unique' : 'pk'] = true;

              table.indexes.push(index);

              break;
            }

            // single column contraint -> update the column instead
            const key = cmd.value.columns[0].value;
            const fieldToUpdate = table.fields.find(f => f.name === key);
            if (fieldToUpdate) fieldToUpdate[cmd.value.unique ? 'unique' : 'pk'] = true;

            break;
          }
```

Reply:
@NQPhuc Ok, I am gonna take your path, I am gonna update this today (and update the test cases)

1. if constraint is on a single column, use "pk"/unique in column (without any name)
2. If constraint is on multiple columns, use "Indexes"

Reply:
@NQPhuc  OK, done your way, this is ready!

Reply:
Thanks, I think this PR is almost done. There's only one thing left: can you help add some test for ALTER TABLE ADD CONSTRAINT with multi-column to make sure it working properly?